### PR TITLE
Extend bundled expo/metro-config

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- Resolves ENG-87
- Basically the same but now with support for `jsx` and any other features we choose to support in the feature.

# How

- ~~Uses the bundled version of `@expo/metro-config` from `expo`~~ waiting until after SDK 41, until then we add a dependency on `@expo/metro-config` in expo-yarn-workspaces.

### CODE REVIEW

File: packages/expo-yarn-workspaces/index.js [hunk 1]
```diff
@@ -62,11 +69,9 @@ exports.createMetroConfiguration = function createMetroConfiguration(projectPath
     },
 
     transformer: {
+      ...defaultConfig.transformer,
       // Ignore file-relative Babel configurations and apply only the project's
       enableBabelRCLookup: false,
-
-      // Temporarily include the Expo asset plugin; figure out a more general way to include it
-      assetPlugins: ['expo/tools/hashAssetFiles'],
```
Comment (ide) [inline packages/expo-yarn-workspaces/index.js:69]: What replaces this?

Comment (giautm) [reply packages/expo-yarn-workspaces/index.js:69]: It's comes from `@expo/metro-config`
https://github.com/expo/expo-cli/blob/master/packages/metro-config/src/ExpoMetroConfig.ts#L34-L42

https://github.com/expo/expo-cli/blob/master/packages/metro-config/src/ExpoMetroConfig.ts#L94

Comment (EvanBacon) [reply packages/expo-yarn-workspaces/index.js:69]: `@expo/metro-config` https://github.com/expo/expo-cli/blob/610c5c60413a2cae824c7b4d25f41244cf71eef8/packages/metro-config/src/ExpoMetroConfig.ts#L36

File: packages/expo-yarn-workspaces/index.js [hunk 2]
```diff
@@ -3,6 +3,7 @@
 const debug = require('debug')('workspaces');
 const findYarnWorkspaceRoot = require('find-yarn-workspace-root');
 const fs = require('fs');
+const { getDefaultConfig } = require('expo/metro-config');
```
Comment (giautm) [inline packages/expo-yarn-workspaces/index.js:6]: Hừm? Missing "At"? `@expo/metro-config` ?

**Review by ide — APPROVED** [review]

**Review by brentvatne — APPROVED** [review]
only concern i have about this is that we can't release a new version of expo-yarn-workspaces until sdk 41 if we use metro-config re-exported from expo. maybe we should just leave it as `@expo/metro-config` for now

**Pull request merged**